Model/AsyncManagement/Charge.php

```diff
@@ -309,13 +310,26 @@ public function authorize($order, $chargeId)
      */
     public function capture($order, $chargeId, $chargeAmount)
     {
+        // Try to load invoice based on charge ID or from the order
         $invoice = $this->loadInvoice($chargeId, $order);
+
+        // Create invoice if we didn't find one for the chargeId but can invoice
         if (!$invoice && $order->canInvoice()) {
             $invoice = $this->invoiceService->prepareInvoice($order);
             $invoice->register();
         }
 
-        if ($invoice && ($invoice->canCapture() || $invoice->getOrder()->getStatus() == Order::STATE_PAYMENT_REVIEW)) {
+        // Finally, load from the order if all else fails
+        if (!$invoice) {
```

Comment:
Should this ever happen, since it was created right above is not found?

Reply:
There is some odd condition I am seeing where the invoice exists on the order already but not associated with the chargeId from above. I don't know off hand why we didn't load it from the order initially, but there are _also_ instances where there is an invoice on the order but the order still has `canInvoice() == true` and we aren't able to capture unless we prepare/register the invoice.

Reply:
weird, since the `loadInvoice` method uses search criteria to find it. It's fine, though, since you've seen it.